### PR TITLE
improve the performances

### PR DESCRIPTION
**Pull request by irevoire**
I removed as much useless characters as possible in the source code:
- This improve the compilation time by at least 0.2% since the parser now doesn't have to skip these characters
- I'm pretty sure it also reduce the size of the final binary
- It also improve the readability of the code. Personally I can see way more code on my screen

image

For  further improvement we could probably merge every files in one single big file.

### CODE REVIEW

File: meilisearch-core/src/criterion/attribute.rs [hunk 1]
```diff
@@ -1,37 +1 @@
...
-            sum_of_attribute
-        }
-
-        let lhs = sum_of_attribute(&lhs.processed_matches);
-        let rhs = sum_of_attribute(&rhs.processed_matches);
-
-        lhs.cmp(&rhs)
-    }
-}
+use std::cmp::Ordering;use slice_group_by::GroupBy;use crate::{RawDocument, MResult};use crate::bucket_sort::SimpleMatch;use super::{Criterion, Context, ContextMut, prepare_bare_matches};pub struct Attribute;impl Criterion for Attribute {fn name(&self) -> &str { "attribute" }fn prepare<'h, 'p, 'tag, 'txn, 'q, 'r>(&self,ctx: ContextMut<'h, 'p, 'tag, 'txn, 'q>,documents: &mut [RawDocument<'r, 'tag>],) -> MResult<()>{prepare_bare_matches(documents, ctx.postings_lists, ctx.query_mapping);Ok(())}fn evaluate(&self, _ctx: &Context, lhs: &RawDocument, rhs: &RawDocument) -> Ordering {#[inline]fn sum_of_attribute(matches: &[SimpleMatch]) -> usize {let mut sum_of_attribute = 0;for group in matches.linear_group_by_key(|bm| bm.query_index) {sum_of_attribute += group[0].attribute as usize;}sum_of_attribute}let lhs = sum_of_attribute(&lhs.processed_matches);let rhs = sum_of_attribute(&rhs.processed_matches);lhs.cmp(&rhs)}}
```
Comment (MarinPostma) [inline meilisearch-core/src/criterion/attribute.rs:1]:
```suggestion
use std::cmp::Ordering;use slice_group_by::GroupBy;use crate::{RawDocument, MResult};use crate::bucket_sort::SimpleMatch;use super::{Criterion, Context, ContextMut, prepare_bare_matches};pub struct Attribute;impl Criterion for Attribute {fn name(&self) -> &str { "attribute" }fn prepare<'h, 'p, 'tag, 'txn, 'q, 'r>(&self,ctx: ContextMut<'h, 'p, 'tag, 'txn, 'q>,documents: &mut [RawDocument<'r, 'tag>],) -> MResult<()>{prepare_bare_matches(documents, ctx.postings_lists, ctx.query_mapping);Ok(())}fn evaluate(&self, _ctx: &Context, lhs: &RawDocument, rhs: &RawDocument) -> Ordering {#[inline]fn sum_of_attribute(matches: &[SimpleMatch]) -> usize {let mut sum_of_attribute = 0;for group in matches.linear_group_by_key(|bm| bm.query_index) {sum_of_attribute += group[0].attribute as usize;}sum_of_attribute}let lhs = sum_of_attribute(&lhs.processed_matches);let rhs =sum_of_attribute(&rhs.processed_matches);lhs.cmp(&rhs)}}
```

this spaca is unecessary, please remove it.

**Review by MarinPostma — CHANGES_REQUESTED** [review]
still many useless spaces. Needs more work.

File: meilisearch-core/src/criterion/attribute.rs (hunk 1 above)
Comment (Kerollmops) [reply meilisearch-core/src/criterion/attribute.rs:1]: Spaces between lifetimes too!

**Comment by irevoire** [comment]
Easter is over, closing.

**Pull request closed without merging**